Files plugin fixes/adaptation

Mingling94 opened the pull request:
- Adapted to new API
- Fixed various bugs (Still addressing many concerns of #175 though)
- Share and shareascii for folders
- Also fixed some deprecated api usage (file.nickname -> file.siapath)
  - Reflect new `renew` attribute in files and file elements
- tools.notType(), function to log improper arguments
  - Check param in each file/folder function
- Removed file.size and file.count getters, trying not to cater file class to folder class. Folder should strictly inherit from file
- Name fields do not take up more space than needed

Note: Many buttons still not working. Much `TODO:`

Mingling94 committed:
Marked TODO:s and updated most API calls

Have to update `/renter/share` and `/renter/shareascii` calls

Mingling94 committed:
Default upload behavior is to autorenew

Mingling94 committed:
Share & shareascii implemented

Share and shareascii for folder puts all inner filepaths into an array

* Also fixed some deprecated api usage (file.nickname -> file.siapath)
  * Reflect new `renew` attribute in files and file elements
* tools.notType(), function to log improper arguments
  * Check param in each file/folder function
* Removed file.size and file.count getters, trying not to cater file
  class to folder class. Folder should strictly inherit from file
*

Mingling94 committed:
Restyled files in the list

DavidVorick commented on plugins/Files/js/file.js:
```diff
@@ -98,42 +108,47 @@ var file = {
+		for (let i = this.parentFolder; i; i = i.parentFolder) {
+			parentFolders.push(i);
+		}
+		return parentFolders.reverse();
```
there's no builtin `split` function?

Mingling94 replied:
This gets references to the folder objects, not the array of folder names (as the function `folderNames()` does)

Mingling94 committed:
Renaming has button, doubleclick downloads files

Mingling94 committed:
browser.downloadSelected() fixed

DavidVorick commented on plugins/Files/js/folder.js:
```diff
@@ -105,26 +77,85 @@ var folder = Object.assign(Object.create(file), {
 	},
 
-	// Share .sia files into folder at destination with same structure
+	// Share .sia files of all contents (deep) to destination
```
this wording is not really clear

Mingling94 committed:
Add file/folder button has greater indication

DavidVorick commented on plugins/Files/js/folder.js:
```diff
@@ -105,26 +77,85 @@ var folder = Object.assign(Object.create(file), {
 
-		// Make folder at destination
-		mkdirp.sync(destination);
+	// Share ascii of all contents (deep)
```
same thing. The word 'deep' really doesn't have any meaning

DavidVorick commented:
You gotta fix the build so it doesn't fail randomly.

Mingling94 commented:
Oh nah this time it's because of linting errors that I neglected to check.

Mingling94 committed:
Renamed 'contents' to 'files'

Mingling94 committed:
Hide action buttons when no selected files

Mingling94 committed:
Updating doesn't clear user selections

Mingling94 committed:
Upload file/folder fix

Mingling94 committed:
Active host prices and count by default

Mingling94 committed:
Use string.includes

Mingling94 committed:
Fixed bugs with refreshing the file list

Mingling94 committed:
Fixed file and folder renaming

folder renaming required that the `this` be preserved when waterfalling
async functions

Mingling94 committed:
Share selected, .sia or ascii, working

Mingling94 commented:
@DavidVorick @lukechampine I think I've addressed all of #175 now...

Mingling94 committed:
Search bar now searches whole library

lukechampine commented on plugins/Files/js/browser.js:
```diff
@@ -226,34 +303,116 @@ var browser = {
+				defaultPath: label,
+			});
 		} else {
+			let totalCount = files.reduce(function(a, b) {
```
woo higher order functions

lukechampine commented on plugins/Files/js/folder.js:
```diff
@@ -48,103 +122,130 @@ var folder = Object.assign(Object.create(file), {
+			pathArray = f.pathArray;
+		} else if (typeof f === 'string') {
+			// File passed in is a file path
+			pathArray = f.split('/');
```
I worry about the use of `/` as the path separator. Could break on Windows.

lukechampine commented on plugins/Files/js/fileElement.js:
```diff
@@ -6,97 +6,115 @@
+	} else if (f.uploadprogress < 100) {
+		detailText = f.uploadprogress.toFixed(0) + '%'; 
+	} else if (f.renewing) {
+		detailText = 'Renewing';
```
`Renewing` makes it sound like something is actively happening, when in reality the file is just set to be renewed as needed. I think in this case, nothing should be displayed at all.
